### ISSUE TITLE
Realtime Pinyin conversion

### ISSUE THREAD

**Issue by AspadaX** [AspadaX/opennote#151]
Convert user input pinyin to Chinese characters as the user types. Give it a debounce.

**Comment by AspadaX** [comment]
As I tested it locally, MiniCPM-V-4.5 turns out to be excellent on converting pinyin in realtime. Whereas, MiniCPM5-1B is totally unusable. 

However, V-4.5 is 9B model, which can be still huge for consumer grade computers. I am thinking if I could finetune the 1B model specifically for pinyin conversion: 

Using Unsloth for finetuning MiniCPM5-1B:
https://github.com/OpenBMB/MiniCPM/blob/main/docs/finetune/unsloth.md

Datasets:
https://cc-cedict.org/editor/editor.php?handler=Download&utm_source=chatgpt.com
https://www.kaggle.com/datasets/duyu09/pinyin-hanzi
https://huggingface.co/datasets/swaption2009/20k-en-zh-translation-pinyin-hsk

**Comment by AspadaX** [comment]
Use case:
1. convert a page of pinyin quick note into Chinese characters

**Comment by AspadaX** [comment]
try finetuning a smaller model for performance: 

https://huggingface.co/Mxode/SmolLM-Chinese-180M
https://unsloth.ai/docs/get-started/fine-tuning-llms-guide/datasets-guide

**Comment by AspadaX** [comment]
what's next
- Use a pinyin IME benchmark
    - Compare against https://github.com/letiantian/Pinyin2Hanzi
- Increase the dataset volume
    - Incorporate Wiki texts and other Chinese sources
- Train with a smaller learning rate

cited
- https://arxiv.org/html/2203.00249v2#S2
- https://github.com/junli-cuny/llm-ime-benchmark

**Comment by AspadaX** [comment]
not very efficient as i experimented. aborted.
